internal/app/singularity/pull.go

```diff
@@ -56,7 +56,7 @@ func LibraryPull(imgCache *cache.Handle, name, fullURI, libraryURI, keyServerURL
 		return fmt.Errorf("could not get image info: %v", err)
 	}
 
-	if noCache {
+	if noCache || imgCache.IsDisabled() {
```

Comment:
Why would one of these be true and not the other? Would't that be a issue if part of the code is detecting noCache, and the other part is not? Unless I missing something, I think there should be a check here, something like:

```go
if noCache != imgCache.IsDisabled() {
    sylog.Debugf("noCache=%v, imgCache.IsDisabled()=%v", noCache, imgCache.IsDisabled())
    return fmt.Errorf("unable to detect if cache is disabled or not")
}
```

**Or**, you can skip the `noCache` bool and only rely on `imgCache.IsDisabled()` for disabling the cache?...

Reply:
Good point! I will check but I am pretty sure you are right, `noCache` most certainly can be skipped.

Reply:
Wait, maybe... If `imgCache.IsDisabled()` is checking if cache is disabled by only looking at the `SINGULARITY_DISABLE_CACHE`, then it will miss the `--disable-cache` command-line-option...

Reply:
Mmmm, good point again. Sounds like we need a few more e2e tests here to make sure everything will be consistent. I will work on the e2e tests in a different PR. We now have enough different code paths that it will be difficult to ensure that we do not break something without these tests.
I may also have a look at the code path the support of `--disable-cache` and make sure things are clean and consistent. I will ping you when code will be ready to get your feedback.

Reply:
Fixed thanks to all the PRs pushed over the past few days related to the cache management. Yay! :)